Product Collection: Implement the Hand-Picked collection

WunderBart opened the pull request:
### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes #50862

This PR adds the Hand-Picked collection to the Product Collection block.

| Picker | Initial (empty) state | w/ picked products | w/ On Sale filter |
| ------------- | ------------- | ------------- | ------------- |
| <img width="648" alt="Screenshot 2024-09-13 at 14 21 45" src="https://github.com/user-attachments/assets/194495e2-8fe4-4989-864e-0661a98b65fb"> | ![image](https://github.com/user-attachments/assets/eb55b6fe-6e6c-4806-a4c5-541ea4ec5d40) | <img width="650" alt="Screenshot 2024-09-13 at 14 23 09" src="https://github.com/user-attachments/assets/499026c9-4d57-4ef6-85ac-fc20d6c3c443"> | <img width="651" alt="Screenshot 2024-09-13 at 14 23 22" src="https://github.com/user-attachments/assets/5095c545-c16a-41d1-980d-cacfa0a1a38e"> |

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

> [!IMPORTANT]
> See [Product Collection: Update heading for Hand-Picked collection](https://github.com/woocommerce/woocommerce/pull/51643#top) for updated testing instructions.

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

- Create a new post.
- Add the Product Collection block and select the Hand-Picked collection.
- Ensure the initial/empty state of the block displays `No products chosen yet. Use "Hand-Picked Products" in Inspector Controls to choose products to display`.
- Ensure the hand-picked control is visible at the top of the block settings panel (see image above).
- Ensure the hand-picked control is unavailable via the `Filters` setting dropdown, when no products are selected.
	- Save the draft and ensure the empty block is not rendered in the front end.
- Add some products in random order via the hand-picked control.
	- Ensure the "as entered" order is preserved in both editor and front end.
- Apply other filters (e.g., on sale) and ensure the results are as expected.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>
<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [x] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->
Product Collection: Implement the Hand-Picked collection
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ObliviousHarmony committed:
Fixed PHP 7.4 Compatibility

The splat operator does not support associative arrays until PHP 8.1.

ObliviousHarmony committed:
Added Related Products Collection Boilerplate

ObliviousHarmony committed:
Added `woocommerceRelatedTo` Collection Parameter

This parameter allows the block to query for products that are related
to the given product IDs.

ObliviousHarmony committed:
Added Collection Related Product Filter

Using `woocommerceRelatedTo`, queries may restrict the products returned
to those that are related to the given product ID(s).

ObliviousHarmony committed:
Removed Unnecessary Tracking

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

ObliviousHarmony committed:
Removing Misunderstood Option

ObliviousHarmony committed:
Linting Fix

ObliviousHarmony committed:
Fixed Test

imanish003 committed:
Merge branch 'trunk' of https://github.com/woocommerce/woocommerce into add/49335-related-products-collection

WunderBart committed:
Initialize the Hand-Picked collection

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

ObliviousHarmony committed:
Removed Unnecessary Keyword

ObliviousHarmony committed:
Support Unlimited `wc_get_related_products()`

This changes related product fetching so that `-1` will return all
related products. I also removed a second +10 offset that seems
to have been accidentally added eight years ago.

WunderBart committed:
Render the collection-specific control

WunderBart committed:
try removing for fixing gh syntax hl

WunderBart committed:
add back

WunderBart committed:
tweak icon render

WunderBart commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/hand-picked.tsx:
```diff
@@ -0,0 +1,102 @@
+
+const Icon = () => {
+	return (
+		<svg
```
The SVG here breaks the syntax highlight in GH, although it doesn't seem to break the code. Anyway, it doesn't feel right as is, but I'm not sure how to handle SVGs in this project since this seems to be the first one that is not available from the [@wordpress/icons package](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library):

<img width="259" alt="Screenshot 2024-09-06 at 13 18 30" src="https://github.com/user-attachments/assets/ff6aed91-f6c3-4180-8878-a84d228d438f">


Any ideas, @imanish003 @kmanijak @xristos3490?

imanish003 replied:
Maybe we can use the same approach as in the [Breadcrumbs block](https://github.com/woocommerce/woocommerce/blob/3d5e7c3610b543db31f4a3adca39597a49a8e4e7/plugins/woocommerce-blocks/assets/js/blocks/breadcrumbs/index.tsx#L31-L38)?  (More such examples can be found in the codebase)
- Separate file for the icon: I don't think it's necessary here. If we will create a new file for icon then probably we will need to create a new folder for icons in Product Collection block.
- `Icon` from `@wordpress/icons` package: I believe we can use it here as well. Here is how it looks like:
```tsx
import { Icon } from '@wordpress/icons';

<Icon
    icon={ handPickedCollectionIcon }
/>
```

What do you think?

WunderBart replied:
> Maybe we can use the same approach as in the [Breadcrumbs block](https://github.com/woocommerce/woocommerce/blob/3d5e7c3610b543db31f4a3adca39597a49a8e4e7/plugins/woocommerce-blocks/assets/js/blocks/breadcrumbs/index.tsx#L31-L38)?

Oh that's perfect - missed this one! Thanks! 🙇

WunderBart replied:
Updated in de596e3e1197b7a8075598383c33c5d07cb32f2b

WunderBart replied:
For some reason, the GH syntax highlighting for this file is still broken. 🙄 🤷

imanish003 replied:
Haha IMO As long as the functionality is correct, we're good to go 🤔

WunderBart commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/hand-picked-products-control.tsx:
```diff
@@ -148,6 +148,34 @@ const HandPickedProductsControl = ( {
+	return (
+		<FormTokenField
+			displayTransform={ transformTokenIntoProductName }
+			label={ __( 'Select Products', 'woocommerce' ) }
```
When the control is displayed as a part of filters, the label should be `Hand-picked products`, and when it's used as the Hand-picked collection-specific control, it should be `Select products`. What would be the best way to contextualize this label? Should we just pass an extra prop?

cc: @imanish003 @kmanijak @xristos3490

imanish003 replied:
I am not sure if we should change it based on this. Maybe we can keep just one of these? For example, Maybe we can use "Hand-picked Products" for both cases? But it's worth confirming with Jill 🙂

If we decide to change it, then I think it's okay to pass an extra prop for now. For example if `label` prop is passed then we use it, otherwise we use the default one. Because `HandPickedProductsControl` is just a reusable React component from my POV.

WunderBart replied:
> I am not sure if we should change it based on this. Maybe we can keep just one of these? For example, Maybe we can use "Hand-picked Products" for both cases? But it's worth confirming with Jill 

Personally, I'd opt for leaving the `Select Products` if we decide to go with one title (see the screenshots below). No strong opinion here, though. @j111q what do you think? Should we keep one title for both scenarios or go with context-specific ones?

| Hand-Picked collection control | Filters drop-down | Filters form |
| ------------- | ------------- | ------------- |
| <img width="283" alt="Screenshot 2024-09-09 at 16 12 19" src="https://github.com/user-attachments/assets/ee6eff4a-e520-4703-a787-a69797a8be99"> | <img width="282" alt="Screenshot 2024-09-09 at 16 12 51" src="https://github.com/user-attachments/assets/92598f02-e6a4-4d4a-adaa-77de16e3e10b"> | <img width="284" alt="Screenshot 2024-09-09 at 16 12 59" src="https://github.com/user-attachments/assets/b576c782-7311-41b7-896b-f64d8ad509f8"> |

kmanijak replied:
I'm totally for the unified label. There's no reason to contextualize which just makes the control more complex (code-wise and UX-wise). `Select products` sounds good to me!

j111q replied:
Hi, it makes sense to me too to use the same label in both contexts 👍 That being said, could we change the copy to `Select specific products` so it's a little clearer? Do you think that makes sense?

WunderBart replied:
Thanks, @j111q! Looking at the other filters now, I wonder if we should adjust the phrasing for a consistent tone, as `Select specific products` would be the only one that calls to action. What if we go with `Selected Products` or `Products Selection` so that it matches other filters?

j111q replied:
Oh that's a great point. Maybe we could go with `Hand-picked Products` then for super consistency? 😆  Consistent structure to other filters, and matches the name of the filter directly.

WunderBart replied:
Following a quick discussion with Jill on Slack, we’ve decided to use `Hand-Picked Products` across both contexts. Additionally, the collection description will be updated to `Select specific products to recommend to customers` to clarify the purpose. 🙌

WunderBart replied:
Updated in https://github.com/woocommerce/woocommerce/pull/51141/commits/a1a23f32d47b4a0c2057077667918c51323110d7.

WunderBart committed:
combine conditions for early return

imanish003 commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/hand-picked.tsx:
```diff
@@ -0,0 +1,102 @@
+		order: 'desc',
+		perPage: 5,
+		pages: 1,
+		timeFrame: {
+			operator: ETimeFrameOperator.IN,
+			value: '-7 days',
+		},
```
I know it's still a draft pull request but wanted to mention that we should remove `timeFrame` from the query 🙇‍♂️

WunderBart replied:
Removed in 6e09d7b480985995626ef9c9e5fe1a60df1cea47

imanish003 commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx:
```diff
@@ -249,9 +254,69 @@ const shouldBeUnmarkedAsConverted = ( props ) => {
+const withCollectionSpecificControls =
+	< T extends EditorBlock< T > >( BlockEdit: ElementType ) =>
+	( props: ProductCollectionEditComponentProps ) => {
+		if ( ! isProductCollection( props.name ) || ! props.isSelected ) {
+			return <BlockEdit { ...props } />;
+		}
```
Nit: I think we can also check for `props.attributes.collection` here instead of inside `CollectionSpecificControls` 🙂

WunderBart replied:
I intentionally moved the current collection check to `CollectionSpecificControls` in case other collections have their own controls in the future so that we're able to selectively render them as we do [here](https://github.com/woocommerce/woocommerce/blob/72112bd5d44918b64344939522139a01fddda556/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx#L164-L190). Does that make sense?

WunderBart committed:
Remove timeFrame from the query

WunderBart committed:
Set control label per context

WunderBart committed:
Use primitives for the icon

WunderBart committed:
Hide 'order by' control for the Hand-Picked collection

WunderBart committed:
Revert "Set control label per context"

This reverts commit cd537befbeaee6a13849592c66d98f1019450721.

WunderBart committed:
Update titles and description

WunderBart committed:
Enable sorting by entered order

j111q reviewed (approved):
copy looks good to me 🙏 thanks bart for the detailed work here!

ObliviousHarmony committed:
Removed `woocommerceRelatedTo` Param

Since we're going to use the collection name, we don't need
this functionality any longer.

ObliviousHarmony committed:
Simplified `post__in` Filtering

Since `merge_queries` was already using an intersection merge for
`post__in`, this second function is unnecessary. I've removed it
as well as refactored the merging logic.

ObliviousHarmony committed:
Pass Collection Name To Final Query

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

WunderBart committed:
Improve the query merging logic

WunderBart committed:
Improve the query merging logic

ObliviousHarmony committed:
Added Custom Collection Handlers

Developers can register collections along with handlers that implement
the custom behavior.

ObliviousHarmony committed:
Removed Second Test Query Construction

As it was, `build_query_vars_from_query_block()` was triggering the
ProductCollection instance hooked into WordPress. After that, we
called `build_frontend_query()` on our test instance. This caused some
weird behavior in tests. As a result, however, the tax_query merge
test couldn't rely on the tax_query transformation done by WordPress.

ObliviousHarmony committed:
Added Related Product Collection Handlers

ObliviousHarmony committed:
Revert "Support Unlimited `wc_get_related_products()`"

This reverts commit 41c837202ebad27031b82f4227cba0b307bc7905.

ObliviousHarmony committed:
Fixed Preview Mode Query

It looks like we were checking the wrong place for the preview
state.

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

WunderBart committed:
Merge branch 'trunk' into add/pc-hand-picked-collection

WunderBart commented:
@imanish003 @kmanijak @xristos3490, what should be the initial state of the collection? Right now, it loads 5 products per the default query/filters. Because hand-picked is a filter and not a whitelist, it will always render products if the filter value is empty. Should we change that behavior and render `no results` until the user hand-picks something?

ObliviousHarmony committed:
Fixed Test

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

ObliviousHarmony committed:
Updated Collection Heading

WunderBart committed:
Handle collection context / no results state

WunderBart commented:
This is ready for another look, as all the functionality is now in place! 🙌
I will add tests next week, likely after the conflicts with https://github.com/woocommerce/woocommerce/pull/51076 are resolved.

cc: @imanish003 @kmanijak @xristos3490 @ObliviousHarmony

WunderBart committed:
Small cleanup

kmanijak commented on plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php:
```diff
@@ -858,29 +870,25 @@ function ( $acc, $query ) {
+				// The post__in query needs special handling, as it should be an
+				// intersection of all post__in queries.
+				if (
+					is_array( $acc['post__in'] ) &&
+					! empty( $acc['post__in'] ) &&
+					is_array( $query['post__in'] ) &&
+					! empty( $query['post__in'] )
```
I'm getting two warnings on the frontend: `Undefined array key "post__in"` from line 879 and 881, I guess we should check if `$acc['post__in']` and `$query['post__in']` exist first.

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/hand-picked.tsx:
```diff
@@ -0,0 +1,90 @@
+		perPage: 5,
+		pages: 1,
+	},
+	hideControls: [ CoreFilterNames.HAND_PICKED ],
```
We should hide `CoreFilterNames.FILTERABLE`. Right now we show "Sync with current query" and I'm not sure that's expected:

EDIT: Added `CoreFilterNames.ORDER` to suggestion as per comment elsewhere. That will hide Order control from Inspector Controls.

```suggestion
	hideControls: [ CoreFilterNames.HAND_PICKED, CoreFilterNames.FILTERABLE, CoreFilterNames.ORDER ],
```

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/hand-picked.tsx:
```diff
@@ -0,0 +1,90 @@
+		'Select specific products to recommend to customers.',
+		'woocommerce'
+	),
+	keywords: [ 'hand-picked products', 'product collection' ],
```
These are redundant, collection will be found with either of these, just because they are part of collection (variation) and original block name. We can think of other keywords like:

```suggestion
	keywords: [ 'specific', 'manual' ],
```

kmanijak commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/index.tsx:
```diff
@@ -85,7 +86,10 @@ const ProductCollectionInspectorControls = (
+		showQueryControls &&
+		shouldShowFilter( CoreFilterNames.ORDER ) &&
+		// See https://github.com/woocommerce/woocommerce/issues/50862
+		attributes.collection !== 'woocommerce/product-collection/hand-picked';
```
Instead of this specific check, please add `CoreFilterNames.ORDER` to `hideControls` in a collection definition.

kmanijak reviewed (dismissed):
Amazing work! I'm so glad to see Handpicked Products collection! 🚀 Here's some feedback:

### More than 5 products

I started adding bunch of products and I thought there's some bugs because at some point they didn't appear in collection. And I realized we limit the display to 5 products per page and single page:

<img width="1152" alt="image" src="https://github.com/user-attachments/assets/82e4762c-7c26-4e18-a5c3-db1992620f45">

This setting makes sense for "fixed" collections where we know we want to limit/control the display. For example, we're deliberately not including pagination for other collections, you get 5 products initially, period. In here it's more in user's hands and I'm wondering how should we address this. Some ideas:
1. Set the `perPage` attribute to `0` so we simply display all of the products. We leave it in user's hands to add pagination or limit `perPage` as they wish
2. Increase `perPage` as user adds more products... But that's stupid, it's basically the same as above but more complex and weird, easy to mess with user's expectations 😅 
3. Set the `pages` attribute to `0` and add Pagination block by default.

I'm thinking we should do no 1. for sure. And I'd do 3. as well to increase discoverability (it's not easy to figure out there's Pagination block: that you can add it, where to put it, etc.).

### Default state

I suggest changing the "No results found" to some other text specifically for this collection. Something like "No products chosen yet. Use "Hand-Picked Products" in Inspector Controls to choose products to display". 

For now, I think I'd simply differentiate text [in here](https://github.com/woocommerce/woocommerce/blob/63d93d73551a1cc908e5385818796c9d042eca2f/plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx#L343) depending on the collection type.

<img width="708" alt="image" src="https://github.com/user-attachments/assets/4219ef36-d3f7-440d-8315-cc2145216f8f">

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

ObliviousHarmony committed:
Better Collection Test Teardown

Passing a callable back means that we can use variables in the
setUp and then the tearDown without any confusion.

ObliviousHarmony committed:
Removed Unnecessary Test Provider

ObliviousHarmony committed:
Fixed `$post__in` Merging

When the intersection is empty it was returning all products. This
makes it so that it returns nothing when there's no results.

kmanijak committed:
Set perPage to 0 and add Pagination block for better discoverability

kmanijak committed:
Improve empty state in Editor by providing meaningful instruction

kmanijak committed:
Make code more defensive

kmanijak committed:
Hide more filters in scope of handpicked through hideFilters API

kmanijak committed:
Merge branch 'trunk' into add/pc-hand-picked-collection

kmanijak reviewed (commented):
I made the changes addressing my feedback from previous round of review hence removing "Request changes". But at the same time I don't feel like I should be approving the PR hence would be good to get a review from someone else!

ObliviousHarmony committed:
Fixed PHPDoc

Co-authored-by: Manish Menaria <the.manish.menaria@gmail.com>

ObliviousHarmony committed:
Hide Empty Related Product Collections

ObliviousHarmony committed:
Merge branch 'trunk' into add/49335-related-products-collection

ObliviousHarmony committed:
Merge branch 'trunk' into add/pc-hand-picked-collection

ObliviousHarmony committed:
Merge branch 'add/49335-related-products-collection' into add/pc-hand-picked-collection

ObliviousHarmony committed:
Removed Duplicate Query Function

ObliviousHarmony committed:
Added Hand-Picked Collection Handler

imanish003 committed:
Revert changes to pnpm

imanish003 commented on plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/inspector-controls/hand-picked-products-control.tsx:
```diff
@@ -158,31 +186,20 @@ const HandPickedProductsControl = ( {
 	);
 };
 
+HandPickedProductsControl.collectionSpecific = true;
+
```
**Question:** What's the purpose of `collectionSpecific`? I can't find any usage of it in the codebase 🤔

imanish003 replied:
CC: @WunderBart

WunderBart replied:
Ah, nice catch - that's just a leftover from something I tested earlier. Will remove.

WunderBart replied:
Removed in https://github.com/woocommerce/woocommerce/pull/51141/commits/f8c4a106eeb0b7a0991e88cba77b1f6e9ba15e30.

imanish003 committed:
Improve comment for empty Hand-Picked collection

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

imanish003 committed:
Merge branch 'add/pc-hand-picked-collection' of https://github.com/woocommerce/woocommerce into add/pc-hand-picked-collection

imanish003 reviewed (changes requested):
Hey @WunderBart,

I tested the changes and everything seems to be working as expected, except for one issue around pagination.

#### Issue: pagination isn't working

I added products such that pagination is visible on Frontend. But It's not working. I suspect, it's because we are setting `perPage` to `0` and `pages` to `0` [here](https://github.com/woocommerce/woocommerce/blob/4e64665c36964d09244e7adbd28371282d227b4f/plugins/woocommerce-blocks/assets/js/blocks/product-collection/collections/hand-picked.tsx#L65-L66).

**expected behavior:**
When I click on 2nd page, I should see products from 2nd page.

**actual behavior:**
No matter which page I click, I always see products from the 1st page.

Here is a demo video:

https://github.com/user-attachments/assets/7161978c-178c-4362-a756-36e79cad1894

imanish003 committed:
Fix: Pagination not working on Frontend

Remove perPage and pages attributes from hand-picked collection

imanish003 commented:
I have approved the PR after:
- Fixing pagination issue
- Updating the PR description
- Adding a changelog entry
- Testing PR again

WunderBart commented on plugins/woocommerce-blocks/assets/js/blocks/product-template/edit.tsx:
```diff
@@ -35,6 +38,16 @@ import './editor.scss';
+const getNoResultsText = ( collection?: CoreCollectionNames | string ) => {
+	if ( collection === CoreCollectionNames.HAND_PICKED ) {
+		return __(
+			'No products chosen yet. Use "Hand-Picked Products" in Inspector Controls to choose products to display',
```
Should the product template be tangled up with a specific collection like that? Maybe we could pass this custom 'no results' text via the Hand-Picked collection attributes or sth? @kmanijak @imanish003

WunderBart replied:
After further inspection, I think we should drop this custom 'no results' message entirely. The reason is that it will also show this message when we hand-pick products but then apply filters that clear the results (no intersection).

As an alternative, I suggest improving the wording of the current message so that it's more informative and matches the wider context so that we can use it for all cases:

**Current message** (displayed for all empty collections):

`No results found.`

**Suggested new message**:

_This would also be displayed for all collections. Choose one, but we can mix and match. I think I like the first one the most!_ 😄 

1. `No products to display. Check your filter settings in the block settings panel.`
2. `No products found. Adjust filters in the block settings to populate this collection.`
3. `No products are currently displayed. Try adjusting the filters in the block settings panel.`

> [!IMPORTANT]
> I used the `settings panel` name from the [block editor handbook](https://developer.wordpress.org/block-editor/) nomenclature. I think we should avoid using `inspector controls` as it's not a user-facing name. 

cc: @j111q

imanish003 replied:
> The reason is that it will also show this message when we hand-pick products but then apply filters that clear the results (no intersection).

Aha! Good catch.

> As an alternative, I suggest improving the wording of the current message so that it's more informative and matches the wider context so that we can use it for all cases:

I really like your suggestion to improve the general 'no results' message. It provides more context and guidance to the user. Here is my suggestion: "No products found. Try adjusting the filters in the block settings panel if there are products in your store."

- It suggests adjusting filters if there are indeed products in the store that aren't showing up due to the current filter settings.
- It implicitly acknowledges that if there are no products in the store at all, adjusting filters won't help.

What do you think?

@WunderBart As we need to merge this PR on Monday, probably we can go with any of the above. If needed, we can iterate on the message in follow-up PR. what do you think?

WunderBart replied:
Thanks for the input, Manish! Two thoughts on the suggested wording:

- For the first part, I think we should go with `No products **to display**`. It matches both situations where filters are and are not applied, while `No products **found**` implies that a search/filtering has already been performed (misleading in the case of the initial Hand-Picked collection state).
- For the second part, I'd skip the `if there are products in your store` bit because it creates a precedent to be followed in all other product-related blocks. I think it should be implied that products must be available to use product-related blocks.

To sum up, I'd go with:

`No products to display. Try adjusting the filters in the block settings panel.`

WDYT?

WunderBart replied:
> As we need to merge this PR on Monday, probably we can go with any of the above. If needed, we can iterate on the message in follow-up PR. what do you think?

As per the above, I went ahead and updated the copy in 6e2ddf447e53c4eaab3770c6bd603b4100dd3c31.

imanish003 replied:
Makes sense to me. Thanks for updating the copy 🙌🏻

WunderBart committed:
Remove unused code

WunderBart committed:
Revert "Improve empty state in Editor by providing meaningful instruction"

This reverts commit 9ac5c20d5e2e23a1cc460a57bcecf19603981754.

WunderBart committed:
Do not apply layout styles for empty results

WunderBart committed:
Improve the wording on the empty results message

WunderBart commented:
The PR tests well for me, @imanish003 @kmanijak. Thanks! 🚀 I'll try adding on some tests here.

imanish003 committed:
Resolve merge conflicts

kimmenbert commented:
Any plans on enhancing this block with custom ordering of handpicked products like mentioned in multiple github issues in the past years. eg: https://github.com/woocommerce/woocommerce/issues/46215#issuecomment-2102760148
